x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go

```diff
@@ -98,6 +98,9 @@ func newFleetGatewayWithScheduler(
 	r fleetReporter,
 	acker fleetAcker,
 ) (*fleetGateway, error) {
+
+	// Backoff implementation doesn't support the using context as the shutdown mechanism.
+	// So we keep a done channel that will be closed when the current context is shutdown.
 	done := make(chan struct{})
```

Comment:
do we need it still?

Reply:
Yeah, backoff doesn't support passing a context to it. so to interrupt the backoff we need to rely on a channel.

Reply:
I've created https://github.com/elastic/beats/issues/18140 in the platform team.